test/vagrant-service-manager/proxy_test.rb

```diff
@@ -0,0 +1,109 @@
+require_relative '../test_helper'
+require 'vagrant/util/downloader'
+
+module VagrantPlugins
+  module ServiceManager
+    describe 'Proxy Test' do
+      before do
+        @machine = fake_machine
+        @ui = FakeUI.new
+        # set test path
+        @plugin_test_path = "#{@machine.env.data_dir}/service-manager/test"
+        ServiceManager.temp_dir = "#{@plugin_test_path}/temp"
+        ServiceManager.bin_dir = "#{@plugin_test_path}/bin"
+
+        @configs = { htpasswd_file: "#{ServiceManager.temp_dir}/.htpasswd" }
+      end
+
+      after do
+        FileUtils.rmtree(@plugin_test_path) if File.directory? @plugin_test_path
+      end
+
+      describe 'Unauthenticated proxy environment' do
+        before do
+          @string_stream = StringIO.new
+          @options = { type: :docker, '--cli-version' => '1.10.0' }
+          @handler = ADBDockerBinaryHandler.new(@machine, @machine.env, @options)
+        end
+
+        after do
+          Thread.kill @download_thread
+          Thread.kill @proxy_thread
+          @proxy_server.shutdown
```

Comment:
Would that even work? You already killed the thread running the proxy. I doubt this will result in a clean shutdown. You most likely don't even need to kill the thread. If you shutdown the server, the thread will exit and then you  just join with the main thread in the tear down method.